tests/monitoring_test.go

```diff
@@ -156,15 +161,140 @@ var _ = Describe("Prometheus Alerts", func() {
+					Name: vmName,
+				},
+				Spec: core.PersistentVolumeSpec{
+					AccessModes: []core.PersistentVolumeAccessMode{
+						core.ReadWriteOnce,
+					},
+					Capacity: core.ResourceList{
+						core.ResourceStorage: resource.MustParse("1Gi"),
+					},
+					PersistentVolumeSource: core.PersistentVolumeSource{
+						CSI: &core.CSIPersistentVolumeSource{
+							Driver: "openshift-storage.rbd.csi.ceph.com",
+							VolumeAttributes: map[string]string{
+								"clusterID":     "test-cluster",
+								"mounter":       "rbd",
+								"imageFeatures": "layering,deep-flatten,exclusive-lock",
+								"mapOptions":    mapOptions,
+							},
+							VolumeHandle: vmName,
+						},
+					},
+				},
+			}
+			Expect(apiClient.Create(ctx, pv)).ToNot(HaveOccurred())
+
+			volumeMode := core.PersistentVolumeBlock
+			pvc = &core.PersistentVolumeClaim{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      vmName,
+					Namespace: strategy.GetNamespace(),
+				},
+				Spec: core.PersistentVolumeClaimSpec{
+					VolumeName: vmName,
+					VolumeMode: &volumeMode,
+					AccessModes: []core.PersistentVolumeAccessMode{
+						core.ReadWriteOnce,
+					},
+					Resources: core.ResourceRequirements{
+						Requests: core.ResourceList{
+							core.ResourceStorage: resource.MustParse("1Gi"),
+						},
+					},
+				},
+			}
+			Expect(apiClient.Create(ctx, pvc)).ToNot(HaveOccurred())
+
+			vmi := NewMinimalVMIWithNS(strategy.GetNamespace(), vmName)
+			vmi.Spec = kubevirtv1.VirtualMachineInstanceSpec{
+				Volumes: []kubevirtv1.Volume{
+					{
+						Name: vmi.Name,
+						VolumeSource: kubevirtv1.VolumeSource{
+							DataVolume: &kubevirtv1.DataVolumeSource{
+								Name: vmi.Name,
+							},
+						},
+					},
+				},
+			}
+			vm = NewVirtualMachine(vmi)
+			vm.Spec.Running = pointer.Bool(false)
+			eventuallyCreateVm(vm)
+		}
+
+		It("Should not fire VirtualMachineCRCErrors", func() {
+			createResources(true)
+			alertShouldNotBeActive("VirtualMachineCRCErrors")
+		})
+
+		It("Should fire VirtualMachineCRCErrors", func() {
+			createResources(false)
+			waitForAlertToActivate("VirtualMachineCRCErrors")
+		})
+	})
 })
 
+func checkAlert(alertName string) *promApiv1.Alert {
+	alerts, err := getPrometheusClient().Alerts(context.TODO())
+	Expect(err).ShouldNot(HaveOccurred())
+	alert := getAlertByName(alerts, alertName)
+	return alert
+}
+
 func waitForAlertToActivate(alertName string) {
-	Eventually(func() *promApiv1.Alert {
-		alerts, err := getPrometheusClient().Alerts(context.TODO())
-		Expect(err).ShouldNot(HaveOccurred())
-		alert := getAlertByName(alerts, alertName)
-		return alert
-	}, env.Timeout(), time.Second).ShouldNot(BeNil())
+	Eventually(func() error {
+		alert := checkAlert(alertName)
+		if alert != nil {
+			return nil
+		}
+		return fmt.Errorf("alert %s not found", alertName)
+	}, env.Timeout(), time.Second).ShouldNot(HaveOccurred())
+}
+
+func alertShouldNotBeActive(alertName string) {
+	Eventually(func() error {
```

Comment:
Can this `Eventually` be dropped and be replaced with the `Consistently` after?

Reply:
not really, `Consistently` would not wait for the alert to disappear and would exit with error immediately